src/app/pages/swap/swap-container.tsx

```diff
@@ -1,71 +1,218 @@
 
-import BtcIcon from '@assets/images/btc-icon.png';
-import XBtcIcon from '@assets/images/xbtc-icon.png';
+import { bytesToHex } from '@stacks/common';
+import { ContractCallPayload, TransactionTypes } from '@stacks/connect';
+import {
+  AnchorMode,
+  PostConditionMode,
+  serializeCV,
+  serializePostCondition,
+} from '@stacks/transactions';
 import BigNumber from 'bignumber.js';
 
-import { createMoney } from '@shared/models/money.model';
+import { logger } from '@shared/logger';
 import { RouteUrls } from '@shared/route-urls';
+import { isDefined, isUndefined } from '@shared/utils';
 
-import { useNativeSegwitBalance } from '@app/query/bitcoin/balance/btc-native-segwit-balance.hooks';
-import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { LoadingKeys, useLoading } from '@app/common/hooks/use-loading';
+import { NonceSetter } from '@app/components/nonce-setter';
+import { defaultFeesMinValues } from '@app/query/stacks/fees/fees.hooks';
+import { useStacksPendingTransactions } from '@app/query/stacks/mempool/mempool.hooks';
+import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useGenerateStacksContractCallUnsignedTx } from '@app/store/transactions/contract-call.hooks';
+import { useSignTransactionSoftwareWallet } from '@app/store/transactions/transaction.hooks';
 
 import { SwapContainerLayout } from './components/swap-container.layout';
 import { SwapForm } from './components/swap-form';
-import { SwapAsset, SwapFormValues } from './hooks/use-swap';
+import { useAlexBroadcastSwap } from './hooks/use-alex-broadcast-swap';
+import { oneHundredMillion, useAlexSwap } from './hooks/use-alex-swap';
+import { useStacksBroadcastSwap } from './hooks/use-stacks-broadcast-swap';
+import { SwapAsset, SwapFormValues } from './hooks/use-swap-form';
 import { SwapContext, SwapProvider } from './swap.context';
 
-// TODO: Remove and set to initial state to 0 with live data
-const tempExchangeRate = 0.5;
+function sortSwappableAssetsBySymbol(swappableAssets: SwapAsset[]) {
+  return swappableAssets
+    .sort((a, b) => {
+      if (a.name < b.name) return -1;
+      if (a.name > b.name) return 1;
+      return 0;
+    })
+    .sort((a, b) => {
+      if (a.name === 'STX') return -1;
+      if (b.name !== 'STX') return 1;
+      return 0;
+    })
+    .sort((a, b) => {
+      if (a.name === 'BTC') return -1;
+      if (b.name !== 'BTC') return 1;
+      return 0;
+    });
+}
+
+function migratePositiveBalancesToTop(swappableAssets: SwapAsset[]) {
+  const assetsWithPositiveBalance = swappableAssets.filter(asset =>
+    asset.balance.amount.isGreaterThan(0)
+  );
+  const assetsWithZeroBalance = swappableAssets.filter(asset => asset.balance.amount.isEqualTo(0));
+  return [...assetsWithPositiveBalance, ...assetsWithZeroBalance];
+}
 
 export function SwapContainer() {
```

Comment:
This is really the `AlexSwapContainer`, when we add magic we'll have another one.

Reply:
At some point we are going to have to address scaling, which I've brought up, how do we know which protocol we are using to handle the swap? Is that chosen by the user in the UI, or are we just detecting which tokens have been selected? There are no designs for this flow atm. cc @markmhendrickson @mica000 once Magic is ready, we will need to know how this should be handled. I think we talked about once the first swap token is selected, we would disable any tokens that can't be swapped with that token by the protocol handling the swap?

I'm not sure just renaming this container makes the code here scale to handle Magic integration as the form itself is nested in the container?

Reply:
I think maybe I need to move some things out specific to alex and have a nested container structure? We might need an `AlexSwapConatiner` and a `MagicSwapContainer` that wraps a `SwapContainer`.